ibis/expr/operations.py

```diff
@@ -2746,6 +2746,27 @@ def output_type(self):
         return rules.shape_like(self.args[0], map_type.value_type)
 
 
+class MapValueOrDefaultForKey(ValueOp):
+
+    input_type = [
+        rules.map(dt.any, dt.any),
+        rules.one_of((dt.string, dt.int_), name='key'),
+        rules.value(name='default')
+    ]
+
+    def output_type(self):
+        map_type = self.args[0].type()
+        value_type = map_type.value_type
+        default_type = self.default.type()
+
+        if not (default_type is dt.null or
```

Comment:
It's okay for this to be `null` since that would be the default return value if a key doesn't exist and `default` is not provided.

Reply:
Oh whoops, nevermind. This is correct.

Reply:
Null checking looks good here.

Small nit: can you write this as `if default_type is not dt.null and value_type != default_type`. It's a little easier to read IMO.

Then will merge.

Reply:
yep.